src/TransitionListAwareTrait.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+namespace Dhii\State;
+
+use Dhii\Util\String\StringableInterface as Stringable;
+use Exception as RootException;
+use InvalidArgumentException;
+use Traversable;
+
+/**
+ * Common functionality for objects that are aware of a list of transitions.
+ *
+ * @since [*next-version*]
+ */
+trait TransitionListAwareTrait
```

Comment:
I feel like this trait, and also `StateListAwareTrait`, have a lot of functionality that doesn't belong in them, and maybe even in this package. I don't see how having a list of transitions and a list of states, and being able to manipulate them, is abstract functionality, because I can't imagine its widespread use.

Reply:
The need that I identified is to store the actual state and transition instances. In a graph-like structure, if the states and transitions are used as keys in a map, then any stringable instances are transformed into strings the moment they are used as keys and the instance information is lost. Keeping a list of the actual instances allows implementations to keep that information.

However, over the weekend, I realized that this isn't how I would implement things, so I have to agree with you here. Instead, what I would do is use a separate graph object that is itself responsible for such things.